lms/djangoapps/courseware/views/views.py

```diff
@@ -1669,7 +1670,18 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True, disable_sta
 
             **optimization_flags,
         }
-        return render_to_response('courseware/courseware-chromeless.html', context)
+
+        try:
+            # .. filter_implemented_name: RenderXBlockStarted
+            # .. filter_type: org.openedx.learning.xblock.render.started.v1
+            context, student_view_context = RenderXBlockStarted.run_filter(
+                context=context, student_view_context=student_view_context
+            )
+        except RenderXBlockStarted.PreventXBlockBlockRender as exc:
```

Comment:
My absolute bad I commented on the [commit](https://github.com/openedx/edx-platform/commit/28e160108218c7f4c85f3fae84eb825d703c3a15#r142615628) and not in the PR which would have been more readable.

Anyways my comment was that:
For a filter is [customary to allow](https://github.com/openedx/edx-platform/blob/f3413fd767d28db4ef9678f284c82a695c881dea/xmodule/vertical_block.py#L178) the user to show something custom.

I think the easiest would be to add a RenderCustomResponse exception such as https://github.com/openedx/openedx-filters/blob/16de16c27a4956211cb21f27a3828e265de1be74/openedx_filters/learning/filters.py#L271.

Then at the platform you can pass the response from the exception to the view.
Such as https://github.com/openedx/edx-platform/blob/59cae77f3600f9e8d4def6f9542d2ae814f4fae2/lms/djangoapps/certificates/views/webview.py#L611

Reply:
Have updated